### PR TITLE
Fix race when registry added after all the previous existing ones hav…

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
…e been run but the running flag not set yet

**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/serviceregistry/aggregate/controller_test.go [hunk 1]
```diff
@@ -518,6 +518,9 @@ type RunnableRegistry struct {
 }
 
 func (rr *RunnableRegistry) Run(stop <-chan struct{}) {
+	if rr.running.Load() {
+		panic("--- registry has been run twice ---")
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/aggregate/controller_test.go:522]: nit: should probably pass a `t` into `runnableRegistry` and use `t.Fatal`?

**Comment by howardjohn** [comment]
lgtm, assigned to Steven as I think they made these changes (maybe not...)

**Review by stevenctl — APPROVED** [review]
lgtm – the odds of hitting this seem extremely small, did you observe it occurring?

**Comment by hzxuzhonghu** [comment]
Yes, I see once in my develop env.

**Pull request merged**